From ef3e06a09d401c75fcd29e09cec5d35cb3f23a66 Mon Sep 17 00:00:00 2001
From: Omri Itach <omrii@marvell.com>
Date: Tue, 19 Apr 2016 13:53:52 +0300
Subject: [PATCH 1114/1240] dts: 80x0: update device tree configurations

armada-80x0.dtsi: A8040 SoC DT
- add inclusion for comphy data
- include apn-806 A0 revision (replace obsolete Z revsision)

armada-80x0-db.dtsi: DB board DT
- align file with current armada-70x0-db_router.dts
- skip setting MPP[0-31]: dedicated for CP0-CP1 connectivity (set as 0xff)
- enable SPI on AP (instead of SPI1 on CP0)

armada-80x0-db-SGMII.dts: DB board - SGMII
- aligned with armada-70x0-db_router.dts

armada-80x0-db-10G.dts: DB board - default board DT:
- aligned with armada-70x0-db_router.dts
- replaced MAC0 (egiga2) via comphy#2 to 10G (KR)

Change-Id: I37527016c3bf33e51fc1ed527dc7afada276a3ef
Signed-off-by: Omri Itach <omrii@marvell.com>
Reviewed-on: http://vgitil04.il.marvell.com:8080/29287
Tested-by: Star_Automation <star@marvell.com>
Reviewed-by: Haim Boot <hayim@marvell.com>
---
 arch/arm/dts/Makefile                 |   7 +-
 arch/arm/dts/armada-80x0-db-10G.dts   |  32 ++++++
 arch/arm/dts/armada-80x0-db-SGMII.dts |  54 ++++++++++
 arch/arm/dts/armada-80x0-db.dts       |  79 --------------
 arch/arm/dts/armada-80x0-db.dtsi      | 196 ++++++++++++++++++++++++++++++++++
 arch/arm/dts/armada-80x0.dtsi         |   5 +-
 arch/arm/dts/cp110-slave.dtsi         |   2 +-
 7 files changed, 291 insertions(+), 84 deletions(-)
 create mode 100644 arch/arm/dts/armada-80x0-db-10G.dts
 create mode 100644 arch/arm/dts/armada-80x0-db-SGMII.dts
 delete mode 100644 arch/arm/dts/armada-80x0-db.dts
 create mode 100644 arch/arm/dts/armada-80x0-db.dtsi

diff --git a/arch/arm/dts/Makefile b/arch/arm/dts/Makefile
index 5e569bf..33fe3da 100644
--- a/arch/arm/dts/Makefile
+++ b/arch/arm/dts/Makefile
@@ -70,8 +70,7 @@ dtb-$(CONFIG_PALLADIUM) += apn-806-pd.dtb \
 			   armada-70x0-pd.dtb
 dtb-$(CONFIG_MVEBU_SPL_DDR_OVER_PCI_SUPPORT) += apn-806-db-dop.dtb \
 						armada-7040-rz-db_dop.dtb
-dtb-$(CONFIG_DEVEL_BOARD) += apn-806-db.dtb \
-			     armada-80x0-db.dtb
+dtb-$(CONFIG_DEVEL_BOARD) += apn-806-db.dtb
 
 ifdef CONFIG_AP806_Z_SUPPORT
 dtb-$(CONFIG_DEVEL_BOARD) += armada-7040-rz-db_nand.dtb \
@@ -86,6 +85,10 @@ dtb-y += armada-70x0-customer0.dtb
 dtb-$(CONFIG_DEVEL_BOARD) += armada-70x0-db_router.dtb 	\
 			     armada-70x0-db_kr.dtb	\
 			     armada-70x0-db_all_pcie.dtb
+
+# A8040 DB device tree files
+dtb-$(CONFIG_DEVEL_BOARD) += armada-80x0-db-10G.dtb \
+			     armada-80x0-db-SGMII.dtb
 endif
 endif #CONFIG_TARGET_ARMADA_8K
 
diff --git a/arch/arm/dts/armada-80x0-db-10G.dts b/arch/arm/dts/armada-80x0-db-10G.dts
new file mode 100644
index 0000000..ad3e1c2
--- /dev/null
+++ b/arch/arm/dts/armada-80x0-db-10G.dts
@@ -0,0 +1,32 @@
+/*
+ * ***************************************************************************
+ * Copyright (C) 2015 Marvell International Ltd.
+ * ***************************************************************************
+ * This program is free software: you can redistribute it and/or modify it
+ * under the terms of the GNU General Public License as published by the Free
+ * Software Foundation, either version 2 of the License, or any later version.
+ *
+ * This program is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+ * GNU General Public License for more details.
+ *
+ * You should have received a copy of the GNU General Public License
+ * along with this program.  If not, see <http://www.gnu.org/licenses/>.
+ * ***************************************************************************
+ */
+
+/dts-v1/;
+
+#include "armada-80x0-db.dtsi" /* include initial Board device tree */
+/ {
+	model = "DB-ARMADA-80x0 (Default)";
+	compatible = "marvell,armada-80x0-db", "marvell,armada-80x0";
+	fdt_config_id = <1>;
+	board_id = <0x13>;
+
+	/*
+	This is the default device tree for Armada 8040-DB board,
+	All board interface settings are reflected in armada-80x0-db.dtsi
+	*/
+};
diff --git a/arch/arm/dts/armada-80x0-db-SGMII.dts b/arch/arm/dts/armada-80x0-db-SGMII.dts
new file mode 100644
index 0000000..39f3947
--- /dev/null
+++ b/arch/arm/dts/armada-80x0-db-SGMII.dts
@@ -0,0 +1,54 @@
+/*
+ * ***************************************************************************
+ * Copyright (C) 2015 Marvell International Ltd.
+ * ***************************************************************************
+ * This program is free software: you can redistribute it and/or modify it
+ * under the terms of the GNU General Public License as published by the Free
+ * Software Foundation, either version 2 of the License, or any later version.
+ *
+ * This program is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+ * GNU General Public License for more details.
+ *
+ * You should have received a copy of the GNU General Public License
+ * along with this program.  If not, see <http://www.gnu.org/licenses/>.
+ * ***************************************************************************
+ */
+
+/dts-v1/;
+
+#include "armada-80x0-db.dtsi" /* include initial Board device tree */
+/ {
+	model = "DB-ARMADA-80x0 (10G)";
+	compatible = "marvell,armada-80x0-db", "marvell,armada-80x0";
+	fdt_config_id = <2>;
+	board_id = <0x13>;
+
+	cp110-master {
+		internal-regs {
+			comphy {
+				/* Serdes Configuration:
+					Lane 0: SGMII2
+					Lane 1: USB3_HOST0
+					Lane 2: SGMII
+					Lane 3: SATA1
+					Lane 4: USB3_HOST1
+					Lane 5: PEX2x1
+				*/
+
+				phy2 {
+					phy-type = <PHY_TYPE_SGMII0>;
+					phy-speed = <PHY_SPEED_1_25G>;
+				};
+
+			};
+			gop {
+				mac0 {
+					phy-mode = "sgmii"; /* lane-2 */
+					status = "okay";
+				};
+			};
+		};
+	};
+};
diff --git a/arch/arm/dts/armada-80x0-db.dts b/arch/arm/dts/armada-80x0-db.dts
deleted file mode 100644
index b908ae5..0000000
--- a/arch/arm/dts/armada-80x0-db.dts
+++ /dev/null
@@ -1,79 +0,0 @@
-/*
- * ***************************************************************************
- * Copyright (C) 2015 Marvell International Ltd.
- * ***************************************************************************
- * This program is free software: you can redistribute it and/or modify it
- * under the terms of the GNU General Public License as published by the Free
- * Software Foundation, either version 2 of the License, or any later version.
- *
- * This program is distributed in the hope that it will be useful,
- * but WITHOUT ANY WARRANTY; without even the implied warranty of
- * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
- * GNU General Public License for more details.
- *
- * You should have received a copy of the GNU General Public License
- * along with this program.  If not, see <http://www.gnu.org/licenses/>.
- * ***************************************************************************
- */
-
-/dts-v1/;
-
-#include "armada-80x0.dtsi" /* include SoC device tree */
-
-/ {
-	model = "DB-ARMADA-80x0";
-	compatible = "marvell,armada-80x0-db", "marvell,armada-80x0";
-	fdt_config_id = <3>;
-	board_id = <0x10>;
-
-	memory {
-		device_type = "memory";
-		reg = <0x00000000 0x10000000 0 0>; /* 256 MB */
-	};
-	ap-806 {
-		internal-regs {
-			i2c0: i2c@511000 {
-				status = "okay";
-			};
-
-			spi0: spi@510600 {
-				status = "okay";
-			};
-		};
-
-		pcie-controller {
-			pcie@1,0 {
-				status = "okay";
-			};
-		};
-	};
-	cp110-master {
-		internal-regs {
-			pinctl@440000 {
-					/*   0 1 2 3 4 5 6 7 8 9 */
-				pin-func = < 0 0 0 0 0 0 0 0 0 0
-					     0 0 0 0 0 0 0 0 0 0
-					     0 0 0 0 0 0 0 0 0 0
-					     0 0 0 0 0 0 0 0 0 0
-					     0 0 0 0 0 0 0 0 0 0
-					     0 0 0 0 0 0 6 6 6 6
-					     0 0 0 0>;
-			};
-		};
-	};
-	cp110-slave {
-		internal-regs {
-			pinctl@440000 {
-					/*   0 1 2 3 4 5 6 7 8 9 */
-				pin-func = < 0 0 0 0 0 0 0 0 0 0
-					     0 0 0 0 0 0 0 0 0 0
-					     0 0 0 0 0 0 0 0 0 0
-					     0 0 0 0 0 0 0 0 0 0
-					     0 0 0 0 0 0 0 0 0 0
-					     0 0 0 0 0 0 6 6 6 6
-					     0 0 0 0>;
-			};
-		};
-	};
-
-};
diff --git a/arch/arm/dts/armada-80x0-db.dtsi b/arch/arm/dts/armada-80x0-db.dtsi
new file mode 100644
index 0000000..9bb6bca
--- /dev/null
+++ b/arch/arm/dts/armada-80x0-db.dtsi
@@ -0,0 +1,196 @@
+/*
+ * ***************************************************************************
+ * Copyright (C) 2015 Marvell International Ltd.
+ * ***************************************************************************
+ * This program is free software: you can redistribute it and/or modify it
+ * under the terms of the GNU General Public License as published by the Free
+ * Software Foundation, either version 2 of the License, or any later version.
+ *
+ * This program is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+ * GNU General Public License for more details.
+ *
+ * You should have received a copy of the GNU General Public License
+ * along with this program.  If not, see <http://www.gnu.org/licenses/>.
+ * ***************************************************************************
+ */
+
+#include "armada-80x0.dtsi" /* include SoC device tree */
+/* TODO:
+#include "armada-80x0-dev-info.dtsi" /* include Marvell specific info (S@R, MPP cmd) */
+
+/ {
+	model = "DB-ARMADA-80x0";
+	compatible = "marvell,armada-80x0-db", "marvell,armada-80x0";
+	fdt_config_id = <1>;
+	board_id = <0x13>;
+
+	ap-806 {
+		internal-regs {
+			spi0: spi@510600 {
+				status = "okay";
+			};
+			i2c0: i2c@511000 {
+				status = "disabled";
+			};
+			pinctl: pinctl@6F4000 {
+					/* MPP Bus:
+						SPI0 [0-3]
+						I2C0 [4-5]
+						UART0 [11,19]
+					*/
+					  /* 0 1 2 3 4 5 6 7 8 9 */
+				pin-func = < 3 3 3 3 3 3 0 0 0 0
+					     0 3 0 0 0 0 0 0 0 3>;
+			};
+		};
+	};
+	cp110-master {
+		internal-regs {
+			spi0@700600 {
+				status = "disabled";
+			};
+			spi1@700680 {
+				status = "disabled";
+			};
+			i2c@701000 {
+				status = "okay";
+			};
+			i2c@211000 { /* MSS_i2c0 */
+				status = "disabled";
+			};
+			i2c@211100 { /* MSS_i2c1 */
+				status = "disabled";
+			};
+			usb3@500000 {
+				status = "okay";
+			};
+			usb3@510000 {
+				status = "okay";
+			};
+			sata@540000 {
+				status = "okay";
+			};
+			mmc@780000 {
+				status = "disabled";
+			};
+			pcie-controller {
+				pcie@0,0 {
+					status = "disabled";
+				};
+				pcie@1,0 {
+					status = "disabled";
+				};
+				pcie@2,0 {
+					status = "okay";
+				};
+			};
+			nand@720000 {
+				status = "disabled";
+			};
+			pinctl@440000 {
+					/* MPP Bus:
+					   [0-31] = 0xff: Keep default CP0_shared_pins:
+						[11] CLKOUT_MPP_11 (out)
+						[23] LINK_RD_IN_CP2CP (in)
+						[25] CLKOUT_MPP_25 (out)
+						[29] AVS_FB_IN_CP2CP (in)
+					   [32,34] SMI
+					   [31]    GPIO: push button/Wake
+					   [35-36] XSMI
+					   [37-38] I2C
+					   [40-41] SATA[0/1]_PRESENT_ACTIVEn
+					   [44-55] RGMII1
+					   [56-62] SD
+					*/
+					/*   0    1    2    3    4    5    6    7    8    9 */
+				pin-func = < 0xff 0xff 0xff 0xff 0xff 0xff 0xff 0xff 0xff 0xff
+					     0xff 0xff 0xff 0xff 0xff 0xff 0xff 0xff 0xff 0xff
+					     0xff 0xff 0xff 0xff 0xff 0xff 0xff 0xff 0xff 0xff
+					     0xff 0    7    0    7    7    7    2    2    0
+					     0    0    0    0    1    1    1    1    1    1
+					     1    1    1    1    1    1    0xE  0xE  0xE  0xE
+					     0xE  0xE  0xE>;
+			};
+
+			comphy {
+				/* Serdes Configuration:
+					Lane 0: SGMII2
+					Lane 1: USB3_HOST0
+					Lane 2: KR (10G)
+					Lane 3: SATA1
+					Lane 4: USB3_HOST1
+					Lane 5: PEX2x1
+				*/
+				phy0 {
+					phy-type = <PHY_TYPE_SGMII2>;
+					phy-speed = <PHY_SPEED_3_125G>;
+				};
+				phy1 {
+					phy-type = <PHY_TYPE_USB3_HOST0>;
+					phy-speed = <PHY_SPEED_5G>;
+				};
+				phy2 {
+					phy-type = <PHY_TYPE_KR>;
+					phy-speed = <PHY_SPEED_10_3125G>;
+				};
+				phy3 {
+					phy-type = <PHY_TYPE_SATA1>;
+					phy-speed = <PHY_SPEED_5G>;
+				};
+				phy4 {
+					phy-type = <PHY_TYPE_USB3_HOST1>;
+					phy-speed = <PHY_SPEED_5G>;
+				};
+				phy5 {
+					phy-type = <PHY_TYPE_PEX2>;
+					phy-speed = <PHY_SPEED_5G>;
+				};
+			};
+			mdio {
+				#address-cells = <1>;
+				#size-cells = <0>;
+				phy2: ethernet-phy@2 {
+					reg = <0>;
+				};
+				phy3: ethernet-phy@3 {
+					reg = <1>;
+				};
+			};
+			gop {
+				mac0 {
+					phy-mode = "kr"; /* lane-2 */
+					status = "okay";
+				};
+				mac2 {
+					phy-mode = "sgmii"; /* lane-0 */
+					phy = <&phy2>;
+					phy-speed = <2500>;
+					status = "okay";
+				};
+				mac3 {
+					phy-mode = "rgmii"; /* rgmii-1 */
+					phy = <&phy3>;
+					status = "okay";
+				};
+			};
+			ppv22@000000 {
+				eth0@010000 {
+					status = "okay";
+				};
+				eth1@020000 {
+					status = "okay";
+				};
+				eth2@030000 {
+					status = "okay";
+				};
+			};
+		};
+	};
+
+	cp110-slave {
+
+	};
+
+};
diff --git a/arch/arm/dts/armada-80x0.dtsi b/arch/arm/dts/armada-80x0.dtsi
index 1cbaafd..3ec1bee 100644
--- a/arch/arm/dts/armada-80x0.dtsi
+++ b/arch/arm/dts/armada-80x0.dtsi
@@ -16,9 +16,10 @@
  * ***************************************************************************
  */
 
-#include "apn-806-db.dtsi"
+#include <dt-bindings/comphy/comphy_data.h>
+#include "apn-806-a0.dtsi"
 #include "cp110-master.dtsi"
-#include "cp110-slave.dtsi"
+/* #include "cp110-slave.dtsi" */
 
 / {
 	compatible = "marvell,armada-80x0";
diff --git a/arch/arm/dts/cp110-slave.dtsi b/arch/arm/dts/cp110-slave.dtsi
index 8d8163d..f148b80 100644
--- a/arch/arm/dts/cp110-slave.dtsi
+++ b/arch/arm/dts/cp110-slave.dtsi
@@ -15,7 +15,7 @@
  * along with this program.  If not, see <http://www.gnu.org/licenses/>.
  * ***************************************************************************
  */
-
+ /* TODO: align with cp110-master.dtsi */
 / {
 	model = "Marvell Armada CPN-110 Slave";
 
-- 
1.9.1

